pkg/controller/route/route_controller.go

```diff
@@ -229,25 +254,65 @@ func (rc *RouteController) reconcile(nodes []*v1.Node, routes []*cloudprovider.R
 		}
 	}
 	wg.Wait()
+
+	// after all routes have been created (or not), we start updating
+	// all nodes' statuses with the outcome
+	for _, node := range nodes {
+		wg.Add(1)
+		nodeRoutes := nodeRoutesStatuses[types.NodeName(node.Name)]
+		allRoutesCreated := true
+
+		if len(nodeRoutes) == 0 {
+			go func() {
```

Comment:
Maybe simpler to just ensure that allRoutesCreated = false in this case, and avoid duplicating the updateNetworkingCondition code

Reply:
We need the duplication to set the nodeReady to false/true if `NewStatus != OldStatus`, The first is when the node has no routes (not ready), the second one is for node that has routes but not all of them created successfully.